src/expr/src/scalar/mod.rs

```diff
@@ -1372,12 +1372,15 @@ impl MirScalarExpr {
                     .collect();
                 // Find an inner operand that occurs more than once, and get a vector of its positions
                 all_inner_operands.sort();
-                let indexes_to_undistribute = all_inner_operands
+                let mut indexes_to_undistribute = all_inner_operands
                     .iter()
                     .group_by(|(a, _i)| a)
                     .into_iter()
                     .map(|(_a, g)| g.map(|(_a, i)| *i).collect_vec())
                     .find(|g| g.len() > 1);
+                // `swap_remove_multiple` cannot handle duplicates
+                indexes_to_undistribute.as_mut().map(|vec| vec.sort());
+                indexes_to_undistribute.as_mut().map(|vec| vec.dedup());
```

Comment:
Nit: Perform sort and dedup in the same closure.

Reply:
done